### PR TITLE
MultiProcessor Implementation

### PR DESCRIPTION
**Pull request by thedjnova**
MultiProcessor Implementation with Scale, Translate, and Opacity Processors

Use SceneProcessor Set's, setting up multiple processors in the sets constructor.

### CODE REVIEW

**Comment by AndreiMisiukevich** [comment]
Hey @thedjnova 
That's amazing! Really great work.

I need some time for review these changes. Then let's merge it and I will refactor code.

File: PanCardView/PanCardView.csproj [hunk 1]
```diff
@@ -50,6 +50,19 @@
     <Compile Include="SceneView.cs" />
     <Compile Include="Processors\BaseSceneBackProcessor.cs" />
     <Compile Include="Processors\BaseSceneFrontProcessor.cs" />
+    <Compile Include="Processors\MultiProcessors\BaseOpacityProcessor.cs" Condition=" '$(EnableDefaultCompileItems)' == 'true' " />
```
Comment (AndreiMisiukevich) [inline PanCardView/PanCardView.csproj]: Could you please remove Condition section?
Because it prevents building app on macbook

**Comment by AndreiMisiukevich** [comment]
SetupAppearingContexts -- can't find this method (

**Comment by AndreiMisiukevich** [comment]
Clamp method not found too =(

File: PanCardView/PanCardView.csproj (hunk 1 above)
Comment (thedjnova) [reply PanCardView/PanCardView.csproj]: Im on High Sierra with VS Mac 7.5.3 and builds alright :/

**Comment by thedjnova** [comment]
Ill check over it all on lunch, I had to pull out some project specific changes that couldn't come over and obviously missed some things.

**Comment by AndreiMisiukevich** [comment]
No problem, thank you for you contributing
I will appreciate if you modify SceneView for using new processors. I will check the sample, which already exist

**Comment by AndreiMisiukevich** [comment]
@thedjnova can you add extension method (Clamp) ?

**Comment by thedjnova** [comment]
Its an Internal Xamarin extension, just pushed a commit that should resolve issues. Also remove the condition, doesn't hurt my builds removing so better to lose the useless config options.

**Comment by AndreiMisiukevich** [comment]
http://www.giphy.com/gifs/1dLRxDQ9VLoBPwLP6U

**Comment by AndreiMisiukevich** [comment]
This is good base for scene feature, but as you can see sometime items blink a little during navigating.
And there is a bug with size of new current element during pan action.

I think, We can merge this PR, but there is a lot of work for refactoring and bugfixing =)

**Comment by thedjnova** [comment]
Thanks, Ill get some time in on it over the weekend.

If its merged in raise the bugs and ill pick them up Saturday (depending on the wife ha).

**Comment by thedjnova** [comment]
@jaspervanmegroot if you adjust the InitialBackPositionPercentage value in the BaseTranslationProcessor then it adjusts where the offscreen cards are. (@AndreiMisiukevich had that in there already)

**Comment by AndreiMisiukevich** [comment]
Ok, cool
I won't merge it until you end with bugs =)
I think it would better if you end with this part, then I will refactor code =)

**Comment by AndreiMisiukevich** [comment]
@thedjnova should I wait some new code improvements? =)

**Comment by thedjnova** [comment]
I don't think I'll be able to get much done until maybe Sunday, so yea hold off on merging until I can have a look.

**Comment by AndreiMisiukevich** [comment]
@thedjnova u r  superstar )

**Comment by AndreiMisiukevich** [comment]
I hope we will figure out with it

**Pull request closed without merging**